lib/ansible/modules/source_control/git.py

```diff
@@ -910,23 +919,38 @@ def switch_version(git_path, module, dest, remote, version, verify_commit, depth
                              stdout=out1, stderr=err1, rc=rc, cmd=cmd)
 
     if verify_commit:
-        verify_commit_sign(git_path, module, dest, version)
+        verify_commit_sign(git_path, module, dest, version, valid_pgpkeys)
 
     return (rc, out1, err1)
 
 
-def verify_commit_sign(git_path, module, dest, version):
+def verify_commit_sign(git_path, module, dest, version, valid_pgpkeys):
     if version in get_annotated_tags(git_path, module, dest):
         git_sub = "verify-tag"
     else:
         git_sub = "verify-commit"
-    cmd = "%s %s %s" % (git_path, git_sub, version)
+    cmd = "%s %s %s --raw" % (git_path, git_sub, version)
     (rc, out, err) = module.run_command(cmd, cwd=dest)
     if rc != 0:
         module.fail_json(msg='Failed to verify GPG signature of commit/tag "%s"' % version, stdout=out, stderr=err, rc=rc)
+    if valid_pgpkeys:
+        fingerprint = get_gpg_fingerprint(err)
+        if fingerprint not in valid_pgpkeys:
+            module.fail_json(msg='Invalid public key "%s"' % fingerprint, stdout=out, stderr=err, rc=rc)
```

Comment:
Maybe not "invalid" but "missing from the whitelist"?

Reply:
Fixed.